### ISSUE TITLE
NC keys: Pasting Nested Content does not generate new keys correctly for subitems

### ISSUE THREAD

**Issue by jbjc** [umbraco/Umbraco-CMS#7133]
Copying and pasting nested content that contains nested content does not work as expected. 

In the back end it seems to work and update fine, however: on the actual site the nested content's nested content seems to be still using the data from the original copied content.

## Umbraco version

I am seeing this issue on Umbraco version: 8.2.2 and 8.3.0

Reproduction
------------

### Bug summary

It seems like pasting nested content that contains nested content only generates a new key for the top level nested content. The embedded nested content that is pasted retains the same key as the original copied content.

### Specifics

No error messages, the content in the backend seems to update and save fine. It is only when viewing the content on the site when you notice the issue. I have looked at the keys for the nested content and the pasted content is definitely using the same key as the original. 

### Steps to reproduce

Create a document type with a nested content picker, this picker should pick a type that contains another nested content picker. This second nested content picker should another type that contains something like a text property.

1. Create the page
2. add the nested content to the page 
3. add nested content (populate the text) to this nested content
4. publish this
5. copy the top level nested content
6. paste the top level nested content 
7. modify the text in the new nested content
8. publish this

The text will appear saved in the backend, however, the keys for the original and copied text will be the same.

![image](https://user-images.githubusercontent.com/57661759/68659775-d0a07580-052f-11ea-8f96-dc60a2bd5c51.png)

(original content was 'test', as you can see the updated content is still showing as the original)
![image](https://user-images.githubusercontent.com/57661759/68659799-d9914700-052f-11ea-8576-b81f3b149243.png)

---
_This item has been added to our backlog [AB#3838](https://umbraco.visualstudio.com/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/3838)_

**Comment by jason721** [comment]
My temporary fix, adding this to the end of the `pasteFromClipboard` function should give everything a new guid. 

```
              function setNewKey(values) {
                    for (var i = 0; i < values.length; i++) {
                        var obj = values[i];
                        if (obj.key) {
                            console.log("Setting key", obj)
                            obj.key = String.CreateGuid();
                            for (var k in obj) {
                               // if this has a property that is an array, we need to check if there's more keys to give a new guid to
                                if (Array.isArray(obj[k])) {
                                    setNewKey(obj[k])
                                }
                            }
                        }
                    }
                }

                for (var t = 0; t < newNode.variants[0].tabs.length; t++) {
                    var tab = newNode.variants[0].tabs[t];
                    for (var p = 0; p < tab.properties.length; p++) {
                        var prop = tab.properties[p];
                        if (prop.view == "nestedcontent") {
                            setNewKey(prop.value);
                        }
                    }
                }
```

**Comment by nul800sebastiaan** [comment]
I'll link this to https://github.com/umbraco/Umbraco-CMS/issues/5913 since I believe they are essentially the same issue.

**Comment by nielslyngsoe** [comment]
Hi @nul800sebastiaan though they both are dealing with "Copy" it does though not seem like the same issue. Excuse me if I'm wrong but this issue here is in regard to copying a single Nested Content item, which is fully a Front-End feature. The linked issue is in regards to Copying a Content Node.

**Comment by nielslyngsoe** [comment]
Working on a PR here: https://github.com/umbraco/Umbraco-CMS/pull/7166

**Comment by warrenbuckley** [comment]
Fixed in 8.6.2
